compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/window/WindowTypeTest.kt

```diff
@@ -0,0 +1,688 @@
+        window?.sendKeyEvent(0, Char(8), KEY_TYPED, KEY_LOCATION_UNKNOWN)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "", selection = TextRange(0), composition = null)
+    }
+
+    @Test
+    fun `q, w, backspace 3x (Chinese, Windows)`() = runTypeTest {
+        // q
+        window?.sendInputEvent("q", 0)
+        window?.sendKeyEvent(81, 'q', KEY_RELEASED)
+        assert(text, "q", selection = TextRange(1), composition = TextRange(0, 1))
+
+        // w
+        window?.sendInputEvent("q'w", 0)
+        window?.sendKeyEvent(87, 'w', KEY_RELEASED)
+        assert(text, "q'w", selection = TextRange(3), composition = TextRange(0, 3))
+
+        // backspace
+        window?.sendInputEvent("q", 0)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "q", selection = TextRange(1), composition = TextRange(0, 1))
+
+        // backspace
+        window?.sendInputEvent(null, 0)
+        window?.sendInputEvent(null, 0)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "", selection = TextRange(0), composition = null)
+
+        // backspace
+        window?.sendKeyEvent(8, Char(8), KEY_PRESSED)
+        window?.sendKeyEvent(0, Char(8), KEY_TYPED, KEY_LOCATION_UNKNOWN)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "", selection = TextRange(0), composition = null)
+    }
+
+    @Test
+    fun `q, w, space, backspace 3x (Chinese, macOS)`() = runTypeTest {
+        // q
+        window?.sendInputEvent("q", 0)
+        window?.sendKeyEvent(81, 'q', KEY_RELEASED)
+        assert(text, "q", selection = TextRange(1), composition = TextRange(0, 1))
+
+        // w
+        window?.sendInputEvent("q w", 0)
+        window?.sendKeyEvent(87, 'w', KEY_RELEASED)
+        assert(text, "q w", selection = TextRange(3), composition = TextRange(0, 3))
+
+        // space
+        window?.sendInputEvent("请问", 2)
+        window?.sendKeyEvent(32, ' ', KEY_RELEASED)
+        assert(text, "请问", selection = TextRange(2), composition = null)
+
+        // backspace
+        window?.sendKeyEvent(8, Char(8), KEY_PRESSED)
+        window?.sendKeyEvent(0, Char(8), KEY_TYPED, KEY_LOCATION_UNKNOWN)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "请", selection = TextRange(1), composition = null)
+
+        // backspace
+        window?.sendKeyEvent(8, Char(8), KEY_PRESSED)
+        window?.sendKeyEvent(0, Char(8), KEY_TYPED, KEY_LOCATION_UNKNOWN)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "", selection = TextRange(0), composition = null)
+
+        // backspace
+        window?.sendKeyEvent(8, Char(8), KEY_PRESSED)
+        window?.sendKeyEvent(0, Char(8), KEY_TYPED, KEY_LOCATION_UNKNOWN)
+        window?.sendKeyEvent(8, Char(8), KEY_RELEASED)
+        assert(text, "", selection = TextRange(0), composition = null)
+    }
+
+    @Test
+    fun `q, w, backspace 3x (Chinese, macOS)`() = runTypeTest {
```

Comment:
the test name mentions the OS (macOS in this case).
But how does the test actually configure the OS? what is the difference in tests for different OS?

Reply:
Seems I see the difference. Every test sends the sequence of key events. Such sequences are different for different OS.

So it's not necessary to specify/configure the OS specific things for the test, beside sending the OS-specifc key event sequence. So these tests may run on any OS.

Reply:
Added a comment in the description of class.

The name of OS is important information.